### PR TITLE
feat: Auto-restart if the configuration file changed

### PR DESCRIPTION
**Pull request by haoel**

### CODE REVIEW

**Comment by samanhappy** [comment]
There is a error log during restarting, is it expected?
`ERRO[2023-04-27T18:11:38+08:00] [Web] HTTP server error: accept tcp [::]:8181: use of closed network connection`

File: conf/conf.go [hunk 1]
```diff
@@ -236,11 +238,61 @@ func getYamlFileFromFile(path string) ([]byte, error) {
...
+		return false
+	}
+
+	// check if the configuration file is modified
+	modified := !bytes.Equal(content, previousYamlFile)
+	previousYamlFile = content
+	return modified
+}
+
+func monitorYamlFile(path string) {
```
Comment (localvar) [inline conf/conf.go:284]:
```suggestion
func monitorYAMLFile(path string) {
```

File: conf/conf.go [hunk 2]
```diff
@@ -236,11 +238,61 @@ func getYamlFileFromFile(path string) ([]byte, error) {
...
 func getYamlFile(path string) ([]byte, error) {
 	if isExternalURL(path) {
-		return getYamlFileFromInternet(path)
+		return getYamlFileFromHTTP(path)
 	}
 	return getYamlFileFromFile(path)
 }
 
+// previousYamlFile is the content of the configuration file
+var previousYamlFile []byte
```
Comment (localvar) [inline conf/conf.go:247]: all `Yaml` should be `YAML`.

```suggestion
var previousYAMLFile []byte
```

**Comment by haoel** [comment]
> 

how to reproduce this?

File: cmd/easeprobe/main.go [hunk 3]
```diff
@@ -222,8 +224,48 @@ func main() {
 		wg.Wait()
 		channel.AllDone()
 		doneSave <- true
-		doneRotate <- true
```
Comment (samanhappy) [inline cmd/easeprobe/main.go:225]: is this remove ok for normal exit?

**Comment by haoel** [comment]
@samanhappy I use the `http.Server.Shutdown()` instead the `Close()`, and deal with the error `ErrServerClosed` , I think this time I use the correct way to shutdown the HTTP Server.

File: cmd/easeprobe/main.go (hunk 3 above)
Comment (haoel) [reply cmd/easeprobe/main.go:225]: good review, delete this accidentally

**Review by samanhappy — APPROVED** [review]

**Review by localvar — APPROVED** [review]

**Pull request merged**